samples/auto_layout.lua

```diff
@@ -0,0 +1,349 @@
+    ["Test Autolayout With Localisation"] = "Test de AutoLayout avec Localisation",
+    ["Third Option"] = "Troisième Option",
+    ["This is long menu text "] = "Ceci est un long texte de menu ",
+    ["This is long text choice "] = "Ceci est un long choix de texte ",
+    ["This is longer option text "] = "Ceci est un texte d'option plus long ",
+}
+
+localization.zh = {
+    ["Action Button"] = "操作按钮",
+    ["Choices"] = "选择：",
+    ["Close"] = "关闭",
+    ["First Option"] = "第一选项：",
+    ["Fourth Option"] = "第四选项：",
+    ["Left Checkbox Option 1"] = "左侧复选框选项1",
+    ["Left Checkbox Option 2"] = "左侧复选框选项2",
+    ["Menu"] = "菜单：",
+    ["Right Three-State Option"] = "右侧三态选项",
+    ["Second Option"] = "第二选项：",
+    ["Short "] = "短 ",
+    ["Test Autolayout With Localisation"] = "自动布局与本地化测试",
+    ["Third Option"] = "第三选项：",
+    ["This is long menu text "] = "这是长菜单文本 ",
+    ["This is long text choice "] = "这是长文本选择 ",
+    ["This is longer option text "] = "这是更长的选项文本 ",
+}
+
+localization.ar = {
+    ["Action Button"] = "زر العمل",
+    ["Choices"] = "الخيارات",
+    ["Close"] = "إغلاق",
+    ["First Option"] = "الخيار الأول",
+    ["Fourth Option"] = "الخيار الرابع",
+    ["Left Checkbox Option 1"] = "خيار المربع الأول على اليسار",
+    ["Left Checkbox Option 2"] = "خيار المربع الثاني على اليسار",
+    ["Menu"] = "القائمة",
+    ["Right Three-State Option"] = "خيار الحالة الثلاثية اليمين",
+    ["Second Option"] = "الخيار الثاني",
+    ["Short "] = "قصير ",
+    ["Test Autolayout With Localisation"] = "اختبار التخطيط التلقائي مع التعريب",
+    ["Third Option"] = "الخيار الثالث",
+    ["This is long menu text "] = "هذا نص قائمة طويل ",
+    ["This is long text choice "] = "هذا خيار نص طويل ",
+    ["This is longer option text "] = "هذا نص خيار أطول ",
+}
+
+localization.fa = {
+    ["Action Button"] = "دکمه عملیات",
+    ["Choices"] = "گزینه ها",
+    ["Close"] = "بستن",
+    ["First Option"] = "گزینه اول",
+    ["Fourth Option"] = "گزینه چهارم",
+    ["Left Checkbox Option 1"] = "گزینه چک باکس سمت چپ 1",
+    ["Left Checkbox Option 2"] = "گزینه چک باکس سمت چپ 2",
+    ["Menu"] = "منو",
+    ["Right Three-State Option"] = "گزینه سه حالته سمت راست",
+    ["Second Option"] = "گزینه دوم",
+    ["Short "] = "کوتاه ",
+    ["Test Autolayout With Localisation"] = "تست آتولایوت با بومی سازی",
+    ["Third Option"] = "گزینه سوم",
+    ["This is long menu text "] = "این متن منوی طولانی است ",
+    ["This is long text choice "] = "این یک انتخاب متن طولانی است ",
+    ["This is longer option text "] = "این متن گزینه طولانی تر است ",
+}
+
+localization.set_locale("fa")
+
+function create_dialog()
+    local dlg = mixin.FCXCustomLuaWindow()
+    dlg:SetTitleLocalized("Test Autolayout With Localisation")
+
+    local line_no = 0
+    local y_increment = 22
+    local label_edit_separ = 3
+    local center_padding = 20
+
+    -- left side
+    dlg:CreateStatic(0, line_no * y_increment, "option1-label")
+        :DoAutoResizeWidth()
+        :SetWidth(0)
```

Comment:
I gather that these methods are coming later, but why can't `DoAutoResizeWidth()` implicitly call `SetWidth(0)`, so that the two don't have to always go together?

Reply:
In real windows we probably don't want to call `SetWidth(0)`. This is more for demonstration. One of the features of `DoAutoResizeWidth` is that it only widens, never narrows. There are many situations where you will want a minimum width, or will want to take the default the control offers as a minimum. (Command buttons are a good example.)